Docs basic agent

ijonele opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

Based on the comprehensive documentation updates, here are the high-level release notes:

- **New Features**
	- Added documentation for building an on-chain AI agent with two order types: Automated Orders and Automated Orders with Price Prediction
	- Introduced Warden Agent Kit documentation with guides for creating and extending AI agents
	- Expanded precompile interaction documentation across various Warden Protocol modules

- **Documentation**
	- Created new sections for building agents, including infrastructure setup, order implementation, and agent capabilities
	- Updated module interaction guides with clearer instructions and references
	- Added quickstart and example guides for Warden Agent Kit

- **Improvements**
	- Refined navigation structure for developer documentation
	- Enhanced clarity of module usage instructions
	- Added links to precompile documentation for easier reference
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

ijonele committed:
created two versions of a plan for the basic agent

ijonele committed:
updated the plan

alijnmerchant21 committed:
Define overview, architecture, user flow and prequsities for basic agent

alijnmerchant21 committed:
Define structure of agent

alijnmerchant21 committed:
Add main contract

Also add description, flow, and safety features

alijnmerchant21 committed:
Update flow

alijnmerchant21 committed:
Update title

alijnmerchant21 committed:
Add Agent factory contract.

Define core functionality and main security features

alijnmerchant21 committed:
Update header

alijnmerchant21 committed:
Implementing the test for Trading Agent System

Implement Price Conditions Test && Mock Uniswap Integration

alijnmerchant21 committed:
Add deployment instructions

alijnmerchant21 committed:
Update order

alijnmerchant21 committed:
Add -  Using the Trading Agent

alijnmerchant21 committed:
Update title

alijnmerchant21 committed:
Improve basic structure

alijnmerchant21 committed:
Improve code for types.sol and IExecution.sol

alijnmerchant21 committed:
Add Registry.sol

alijnmerchant21 committed:
Add RLP Encode Library

alijnmerchant21 committed:
Add implementation for precompile functions

alijnmerchant21 committed:
Code update - src/BasicOrder.sol

Added flow, key features and main components

Change order

alijnmerchant21 committed:
Code change - Order Factory

Add Key features - Factory pattern, management, order creation

Change order

alijnmerchant21 committed:
Add code - deploy and create order script

Add brief explanation

New order

alijnmerchant21 committed:
Update .env file

Demo how to deploy base contract

Create a new order + explain parameter

Update verify deployment

Add Monitoring and Managing Orders

Change order

alijnmerchant21 committed:
Remove testing

alijnmerchant21 committed:
Update architecture + Add description to directory structure

alijnmerchant21 committed:
Replace full code with instances + link to code.

Strip down lot of verbose

alijnmerchant21 committed:
Update title

Add code reference

Add security features

alijnmerchant21 committed:
Replace full code with instances + link to code.

Strip down lot of verbose

alijnmerchant21 committed:
Replace full code with instances + link to code.

Strip down lot of verbose

alijnmerchant21 committed:
Fixes to the parameters

alijnmerchant21 committed:
Add some changes to precompile code

Add integration testing

Add test scenarios

Add usage script

Add reference to mock and Test

ijonele committed:
partially edited the basic agent section

ijonele committed:
minor fixes in mock precompiles

ijonele committed:
more editing

ijonele committed:
finalized almost everything

ijonele committed:
edited the BasicOrder article

ijonele committed:
a small fix

ijonele committed:
edited the introduction and prerequisites, finalized everything

ijonele committed:
resolved a conflict

ijonele committed:
configured the left menu

ijonele committed:
deleted mentions of the trading agent

ijonele committed:
a minor fix

ijonele committed:
a minor fix

ijonele committed:
new structure

ijonele committed:
new structure

ijonele committed:
edited the infrastructure part and the main intro

ijonele committed:
edited the narrative about basic orders

ijonele committed:
fixed links

alijnmerchant21 committed:
Docs agent overhaul (#1180)

* Add warden knowledge base

Solidity knowledge base

Improve overall structure

* Minor fixes

* Minor fixes

* Minor fixes

* Minor fixes

* Add overview

Add learning path

* Minor fixes

* Update code

Add architecture

Update components

Add extension points

Update security considerations

* Update overview

Split the code into more understandable functions

Add extension points

* Update overview

Add steps for setup

Add Deployment steps

Add Monitoring & Management

Add Troubleshooting Guide

* Add introduction and preq

* Clean slate for price prediction

alijnmerchant21 committed:
Docs advance agent overhaul (#1182)

* Update intro

Add overview

Add key enhancements

Add Technical Prerequisites

Add core components

Add implmentation plan

Add key differences

* Update reference to infra for orders

* Add core components

Add Initialize Prediction System

Add Implement Enhanced Price Monitoring

Add Price Normalization Utilities

Add Enhanced Price Conditions

Add Testing Advanced Features

Key Security Considerations

* Added Core Components

Added Enhanced Order Creation

Added Order Data Validation

Added Enhanced Address Computation

Added Factory Testing

Added Security Considerations

* Add enviornment setup

Add Deploy order

Add Monitor status and events

Add troubleshooting

Add Security considerations

Add next steps

* Add summary

ijonele committed:
cosmetic changes (#1188)

ijonele committed:
fixed broken links

ijonele committed:
editing intro

ijonele committed:
editing intro

ijonele committed:
Merge remote-tracking branch 'origin/main' into docs-basic-agent

ijonele committed:
prerequisites + cross-links to precompiles and modules

ijonele committed:
stylistic features

ijonele committed:
improved the consistency of terminology + mminor fixes

ijonele committed:
fixed links

ijonele committed:
edited introductuions

ijonele committed:
edited basic orders

ijonele committed:
finished editing basic orders

ijonele committed:
fixed the price condition code in the basic agent

ijonele committed:
edited articles about deployment

ijonele committed:
edited articles about implementing and creating advanced orders

ijonele committed:
a minor fix in the deployment articles

ijonele committed:
improved summary

ijonele committed:
fixed links

ijonele committed:
fixed a link in the env config

alijnmerchant21 committed:
Merge branch 'main' into docs-basic-agent

alijnmerchant21 commented on docs/developer-docs/docs/build-an-agent/build-an-onchain-ai-agent/prerequisites.md:
```diff
@@ -0,0 +1,108 @@
+---
+sidebar_position: 1
```
@ijonele shouldn't the position be 2?

ijonele replied:
fixed this

alijnmerchant21 commented on docs/developer-docs/docs/build-an-agent/build-an-onchain-ai-agent/build-the-infrastructure-for-orders/create-helpers-and-utils.md:
```diff
@@ -0,0 +1,258 @@
+---
+sidebar_position: 2
```
@ijonele this should be 1 as it is the first sub category?

ijonele replied:
fixed the positions in the entire subsection

alijnmerchant21 commented on docs/developer-docs/docs/build-an-agent/build-an-onchain-ai-agent/implement-automated-orders/implement-orders.md:
````diff
@@ -0,0 +1,267 @@
+  if (_executed) revert ExecutedError();
+  ```
+
+## Extension points
````
I'm not sure if we should have an extension point here. Shouldn't it be only in advance order?

ijonele replied:
I thought you added extension points to basic orders to show how to turn them into advanced orders. Do you think we don't actually need them?

alijnmerchant21 commented on docs/developer-docs/docs/build-an-agent/build-an-onchain-ai-agent/summary.md:
```diff
@@ -0,0 +1,116 @@
+
+## What's next?
+
+Please note that these Orders are just examples of what you can build with Agents. Sky is the limit! If you have an interesting idea, make a PR to the [repository with our example](https://github.com/warden-protocol/wardenprotocol/tree/main/solidity/orders).
```
Please note that these Orders are just examples of what you can build with Agents. The sky is the limit! If you have an interesting idea, make a PR to the [example repo](https://github.com/warden-protocol/agent-kit-examples).

ijonele replied:
fixed

alijnmerchant21 reviewed (commented):
Great job @ijonele this looks in pretty good shape. I wanted to make a couple of points:

1. Have you double-checked the code with the corresponding file on GitHub?
2. Do we need the extension verbose in the basic contract? (Looks ok to me, but asking to be sure)
3. Can you add the following verbose to the tutorial, at the start maybe?
```
1. We generalised the order. You can send an order to any application. It doesn't have to be Uniswap. You can send a transaction to any Ethreum app. From sending eth to calling any arbitrary contract.

2. The order contract, supports arbitrary transactions on Ethereum and any EVM L2. They have the same tx scheme. So you can also use any application on any EVM L2.

3. The token prediction model is a model that we've created. It's a placeholder. We are working on letting smart contracts consume arbitrary HTTP requests.
```
4. I have only reviewed basic and advance orders. Not reviewed other file changes like precompiles, glossary, etc.

ijonele committed:
updates after the review

ijonele commented:
> Great job @ijonele this looks in pretty good shape. I wanted to make a couple of points:
> 
> 1. Have you double-checked the code with the corresponding file on GitHub?
> 2. Do we need the extension verbose in the basic contract? (Looks ok to me, but asking to be sure)
> 3. Can you add the following verbose to the tutorial, at the start maybe?
> 
> ```
> 1. We generalised the order. You can send an order to any application. It doesn't have to be Uniswap. You can send a transaction to any Ethreum app. From sending eth to calling any arbitrary contract.
> 
> 2. The order contract, supports arbitrary transactions on Ethereum and any EVM L2. They have the same tx scheme. So you can also use any application on any EVM L2.
> 
> 3. The token prediction model is a model that we've created. It's a placeholder. We are working on letting smart contracts consume arbitrary HTTP requests.
> ```
> 
> 4. I have only reviewed basic and advance orders. Not reviewed other file changes like precompiles, glossary, etc.

@alijnmerchant21 

1. I checked the code in some occasions, not everywhere, since you updated the docs recently and theoretically the code shouldn't have changed.
2. I think the extension part is excessive. I actually already deleted it in the deployment part. However, let's keep the rest just in case. We can delete it later when we continue polishing the docs after the hackathon.
3. I updated the section intro with the info you provided, please review it. Look at the tip and the note I added. I didn't understand what "placeholder" means in this context, so I skipped it for now. See /build-an-onchain-ai-agent/introduction

ijonele committed:
Merge branch 'docs-basic-agent' of https://github.com/warden-protocol/wardenprotocol into docs-basic-agent

alijnmerchant21 commented on docs/developer-docs/docs/build-an-agent/build-an-onchain-ai-agent/implement-automated-orders-with-price-prediction/introduction.md:
:::note
 The price prediction system is a solution implemented by Warden. We're working on enabling smart contracts to consume arbitrary HTTP requests.
 ::: 
 
 - We should say : the prediction model is an example of what can be built. x/async opens up other avenues of off-chain computation with on-chain verification limited only by users imagination.

alijnmerchant21 commented:
Thank you @ijonele 

> 3. I updated the section intro with the info you provided, please review it. Look at the tip and the note I added. I didn't understand what "placeholder" means in this context, so I skipped it for now. See /build-an-onchain-ai-agent/introduction

Place holder meant that the prediction model is not a perfect model out of the box. This is to show the capabilities of x/async.

ijonele committed:
updated the price predicition note

jjheywood committed:
feat(docs): added agent kit docs (#1197)

* feat(docs): added agent kit docs

* feat(docs): updated core actions

* feat(docs): added custom tool example and cli-agent example

* feat(docs): added custom tool example and cli-agent example

ijonele committed:
added the main intro

ijonele committed:
Merge branch 'main' into docs-basic-agent

ijonele committed:
onchain in the main intro
